### PR TITLE
Don't offer Use Explicit Type for target typed new

### PR DESCRIPTION
**Pull request by davidwengier**
Noticed this today while doing something else. Couldn't find a bug for it, should I log one?

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/TypeStyle/CSharpUseImplicitTypeHelper.cs [hunk 1]
```diff
@@ -286,6 +286,16 @@ private static bool IsSafeToSwitchToVarWithoutNeedingSpeculation(DeclarationExpr
                 return false;
             }
 
+            // var cannot be used with target typed new
+#if CODE_STYLE
```
Comment (davidwengier) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/TypeStyle/CSharpUseImplicitTypeHelper.cs:290]: I have no idea if this is the right thing to do with `CODE_STYLE`, or really what `CODE_STYLE` is or means, but its Friday afternoon for me so thought I'd just put up a PR and see what happens :D

Comment (mavasani) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/TypeStyle/CSharpUseImplicitTypeHelper.cs:290]: Looks good to me. @sharwell for another pair of eyes.

Comment (sharwell) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/TypeStyle/CSharpUseImplicitTypeHelper.cs:290]: There is no need for the condition here. Just use `SyntaxKindEx` on both paths.

Comment (sharwell) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/TypeStyle/CSharpUseImplicitTypeHelper.cs:290]: ➡️ #46489

**Pull request merged**